migrations/lib/converters/discourse/steps/group_users.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Migrations::Converters::Discourse
+  class GroupUsers < ::Migrations::Converters::Base::ProgressStep
+    attr_accessor :source_db
+
+    def max_progress
+      @source_db.count <<~SQL
+        SELECT COUNT(*)
+        FROM group_users
+        WHERE user_id >= 0
+      SQL
+    end
+
+    def items
+      @source_db.query <<~SQL
+        SELECT *
+        FROM group_users
+        WHERE user_id >= 0
```

Comment:
Depending on what we decide to do with automatic groups, we might need to filter automatic groups in here.

Reply:
My current thinking around this is that we still convert the automatic groups and their related entities (`group_users` in this case). During import, we can then map these automatic groups to the destination instance’s corresponding automatic groups  without importing them as new. And then ensure `group_users` end up in the right automatic group.

 So, for example, a moderator on the source site would automatically become a moderator on the destination site.

The unanswered question, here though, is what to do if the automatic group membership criteria differ between instances, such that a user who was eligible in the source site becomes ineligible on the destination site?  I'm reviewing the core implementation.

Reply:
Yes, definitely look at the core implementation to figure out what's the best course of action. I'm not familiar with that logic either. But I think it might make sense to exclude the TL groups.